### DISCUSSION TITLE
Factorization caching

### DISCUSSION THREAD

**Discussion by charleskhazoom** [osqp/osqp#570] [General]
Can OSQP do factorization caching of the KKT system if only parts of the $P$ and $A$ and rho matrices change when solving a new QP ? Or does factorization caching only work when vectors $q$, $l$ and $u$ change ?

**Comment by goulart-paul** [comment]
Factorization caching only works in OSQP if P and A do not change.   If you change an element of either then you need to completely refactor the KKT system.   

Note that by "completely refactor" I mean only the numerical factorisation.   If you change elements of P and A, but only in places that were already in structural nonzeros, then you still make a saving on the symbolic factorisation and memory allocation phases since the factorisation of the KKT matrix keeps the same sparsity pattern.
